soc/xtensa/espressif_esp32/esp32/power.c

```diff
@@ -21,17 +21,6 @@ void pm_state_set(enum pm_state state, uint8_t substate_id)
 	ARG_UNUSED(substate_id);
 
 	switch (state) {
-	case PM_STATE_SOFT_OFF:
```

Comment:
> I haven't seen cases where applications want the policy manager to decide when to power off the system based on timing constraints

One thing that we need to account is, we allow applications to define their own policy based on their own constraints / use cases. I don't see a reason to remove it, maybe having a way to exclude it from the "default" policy manager. Still, that is a valid power state and a custom policy manager could use it. To avoid duplication we could have it in a common place that is used in both code paths.